homeassistant/components/soma/config_flow.py

```diff
@@ -0,0 +1,58 @@
+"""Config flow for Soma."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST, CONF_PORT)
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 3000
+
+
+class SomaFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Instantiate config flow."""
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        if user_input is None:
+            data = {vol.Required(CONF_HOST): str,
+                    vol.Required(CONF_PORT, default=DEFAULT_PORT): int
+                    }
+
+            return self.async_show_form(
+                step_id='user',
+                data_schema=vol.Schema(data)
+            )
+
+        return await self.async_step_creation(user_input)
+
+    async def async_step_creation(self, user_input=None):
+        """Finish config flow."""
+        from api.soma_api import SomaApi
+        api = SomaApi(user_input['host'], user_input['port'])
+        await self.hass.async_add_executor_job(
+            api.list_devices)
+        _LOGGER.info('Successfully set up Soma Connect')
```

Comment:
If the connection doesn't succeed we should pass an error to the user form.

Reply:
I added a try/except around that but for some reason trying to catch ConnectionError or ConnectionRefusedError didn't work and so I had to just catch all exceptions. It works but it's not nice like this.

Reply:
What error are we getting?

Reply:
ConnectionError doesn't catch the exeption - but when I catch all and print out the exception object it says that it's a ConnectionError type. I'm not familiar enough with Python to debug why this seems to happen - maybe something to do with the async execution.

Reply:
Try catching `RequestException` and then if that doesn't work try `OSError`.

Reply:
This seems to also work, thank you.